c7n/resources/sns.py

```diff
@@ -119,3 +119,81 @@ def process_resource(self, client, resource):
         return {'Name': resource['TopicArn'],
                 'State': 'PolicyRemoved',
                 'Statements': found}
+
+
+@SNS.action_registry.register('modify-statements')
+class ModifyPolicyStatement(ModifyPolicyBase):
+    """Action to modify policy statements from SNS
+
+    :example:
+
+    .. code-block:: yaml
+
+           policies:
+              - name: sns-cross-account
+                resource: sns
+                filters:
+                  - type: cross-account
+                actions:
+                  - type: modify-statements
+                    add-statements: [statement]
+                    remove-statements: [statement] or *
+    """
+
+    permissions = ('sns:SetTopicAttributes', 'sns:GetTopicAttributes')
+
+    def process(self, resources):
+        replace = False
+        results = []
+        client = local_session(self.manager.session_factory).client('sns')
+        additions = self.data.get('add-statements', [])
+        deletions = self.data.get('remove-statements', [])
+
+        if deletions == "*":
+            replace = True
+
+        for r in resources:
+            new_policy = {u'Version': u'2012-10-17', u'Statement': []}
+            old_policy = r.get('Policy') or '{}'
+            old_policy = json.loads(old_policy)
+
+            if replace:
+                new_policy = self.process_replace(old_policy, r)
+            if len(additions) and len(deletions) and not replace:
+                new_policy['Statement'] += self.process_add_and_delete(old_policy, r)
+            if len(additions) and not len(deletions) and not replace:
+                new_policy['Statement'] += self.process_addition(old_policy, r).get('Statement')
+            if not len(additions) and len(deletions) and not replace:
+                new_policy['Statement'] += self.process_deletion(old_policy, r)
+            results += {
+                'Name': r['TopicArn'],
+                'State': 'PolicyModified',
+                'Statements': new_policy
+            }
+            client.set_topic_attributes(
+                TopicArn=r['TopicArn'],
+                AttributeName='Policy',
+                AttributeValue=json.dumps(new_policy)
+            )
+        return results
+
+    def process_add_and_delete(self, old_policy, resource):
+        new_policy = self.add_statements(old_policy, resource)
+        new_policy, found = self.remove_statements(
+            old_policy, resource, CrossAccountAccessFilter.annotation_key)
+        return new_policy
+
+    def process_addition(self, old_policy, resource):
+        new_policy = self.add_statements(old_policy, resource)
+        if old_policy == new_policy:
```

Comment:
you should be able to just 

```python
return self.add_statements(old_policy, resource)
```

doing a check on `if old_policy == new policy: return old_policy` and returning `new_policy` in the end do they same thing